manifests/init.pp

```diff
@@ -1,29 +1,122 @@
-class rngd  (
-  $config            = $rngd::params::config,
-  $config_template   = $rngd::params::config_template,
-  $extra_options     = $rngd::params::extra_options,
-  $package_ensure    = $rngd::params::package_ensure,
-  $package_name      = $rngd::params::package_name,
-  $service_enable    = $rngd::params::service_enable,
-  $service_ensure    = $rngd::params::service_ensure,
-  $service_manage    = $rngd::params::service_manage,
-  $service_name      = $rngd::params::service_name,
-) inherits rngd::params {
-  
-  validate_absolute_path($config)
+# == Class: rngd
+#
+class rngd (
+  $config_file     = 'USE_DEFAULTS',
+  $config_template = 'rngd/rngd.erb',
+  $extra_options   = undef,
+  $package_ensure  = 'present',
+  $package_name    = 'USE_DEFAULTS',
+  $service_enable  = true,
+  $service_ensure  = 'running',
+  $service_manage  = true,
+  $service_name    = 'USE_DEFAULTS',
+  # deprecated variables
+  $config          = undef,
```

Comment:
Could you please send a PR to remove all the deprecated bits. We aren't interested in maintaining backward compatibility since we are going from a < 1.0.0 release to 1.0.0